### PR TITLE
Keccak + SHA3

### PR DESCRIPTION
**Pull request by maciejhirsz**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -5,7 +5,7 @@ homepage = "https://github.com/multiformats/rust-multihash"
 
 keywords = ["multihash", "ipfs"]
 
-version = "0.4.2"
+version = "0.4.3"
```
Comment (dignifiedquire) [inline Cargo.toml]: Can you remove this? I like to make version bumps its own commit, so I can tag it.

File: tests/lib.rs [hunk 2]
```diff
@@ -14,45 +14,55 @@ fn hex_to_bytes(s: &str) -> Vec<u8> {
...
-        Hash::SHA2256
-    );
-
-    let hash: Vec<u8> = encode(Hash::SHA2512, b"helloworld").unwrap();
-    assert_eq!(
-        decode(&hash).unwrap().alg,
-        Hash::SHA2512
+    assert_decode!(
+        SHA1, SHA2256, SHA2512, SHA3224, SHA3256, SHA3384, SHA3512,
+        Keccak224, Keccak256, Keccak384, Keccak512
```
Comment (dignifiedquire) [inline tests/lib.rs]: while you are in here, could you add a roundtrip macro test as well please?

Comment (maciejhirsz) [inline tests/lib.rs]: decoding was already a roundtrip test, added new ones for decoding from a predefined hashes instead

**Comment by dignifiedquire** [comment]
🎉

**Pull request merged**